### ISSUE TITLE
Add an "after" operator

### ISSUE THREAD

**Issue by All-StarsPixel** [godotengine/godot-proposals#7964]
### Describe the project you are working on

an Undertale fangame

### Describe the problem or limitation you are having in your project

Then diagonals are a pain to add because the animations depend on which input came first. I'm trying to use an array for this when an "after" operator could make this a lot less complicated

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

it would be so much easier than using arrays to get the same result. With arrays it can ruin the framerate and be hard to understand, but with an after, it would be easy to make, fix and read while making your code less convoluted.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

if Input.get_action_strength("ui_up") after Input.get_action_strength("ui_right"):
     animation.play("Walk Right")

### If this enhancement will not be used often, can it be worked around with a few lines of script?

you can use an array, but it would be a pain to make and would make your code messy.

### Is there a reason why this should be core and not an add-on in the asset library?

I feel like it would be nice for people with less experience. The asset librarby isn't the first place you would look and this feels like the type of thing that should just be here by default.

**Comment by BtheDestroyer** [comment]
I'm confused by what the intended behavior is? What's the work-around to achieve the desired result with currently available tools?

**Comment by theraot** [comment]
I wonder if this is sufficient for you:

```
if Input.is_just_action_pressed("ui_up") and Input.is_action_pressed("ui_right") and not Input.is_action_just_pressed("ui_right"):
    animation.play("Walk Right")
```

What I imagine would be difference is that this would only enter one frame, instead of every frame while these actions are pressed. It might be sufficient as long as the code you have is idempotent, but it might also affect the code flow you have.

If that is sufficient, I would suggest the following: Adding a new `is_action_held`:

```
if Input.is_just_action_pressed("ui_up") and Input.is_action_held("ui_right"):
    animation.play("Walk Right")
```

If that is not sufficient, then I suggest adding a new `get_action_press_frame` (and there would be an accompanying `get_action_release_frame`):

```
if Input.is_action_pressed("ui_up") and Input.is_action_pressed("ui_right") and Input.get_action_press_frame("ui_up") > get_action_press_frame("ui_right"):
    animation.play("Walk Right")
```

Which perhaps we could simplify to a `is_action_pressed_after` (this a better name, since the action pressed first is held):

```
if Input.is_action_pressed_after("ui_up", "ui_right"):
    animation.play("Walk Right")
```

Perhaps `are_action_held_sequentially` (I still don't like it because the action pressed second might not be held more than a frame)?

```
if Input.are_action_held_sequentially("ui_right", "ui_up"):
    animation.play("Walk Right")
```

Anyway, tell me if anything of what I said makes sense for your use case.

**Comment by BtheDestroyer** [comment]
I'd be in favor of an `Input.get_action_press_frame` and `Input.get_action_release_frame`. 

If I'm understanding correctly based on @theraot's suggestion, an `after` operator which evaluates as `true` only when the first operand evaluated as `true` and second operand evaluated as `false` in prior evaluations and both operands now evaluate as `true`.

My issue with something that broad is that it could lead to wasted memory and runtime in the best case (needing to constantly evaluate conditions which _could_ be used in `after` statements) and lead to user confusion in the worst case.

**Comment by JoNax97** [comment]
I don't buy that this needs to be added in any form. Keeping track of each input's frame is pretty niche and it can be done in under 20 lines of code. 

Contrary to OP's claims, If done correctly it should not cause any performance problems.

**Comment by Mickeon** [comment]
I agree with the above. There's many reasons not to "memorize action input times" as a built-in feature:
- Most actions do not need to store this information.
- The definition of "input time" may vary on a case-by-case basis. Sometimes it's in actual seconds. Sometimes, all that's needed is the frames passed between inputs.
- What if an input of the same action group is pressed, while the action is already pressed? How to handle this?

If someone has actual concrete examples to help this users and anyone looking for a similar solution, feel free to share. A wholy new _operator_ for something like this is quite silly, however.

**Comment by theraot** [comment]
Thinking about this, here is another idea: `get_pressed_frames`, returning an `int` like this:

- (negative): never.
- `0`: It is not pressed.
- `1`: It is just pressed (pressed between the last and this frame).
- `n` (greater than `1`): It has been pressed for multiple frame (it is held).

Then, you could write:

```
if (
    Input.is_action_pressed("ui_up")
    and Input.get_pressed_frames("ui_right") > Input.is_just_action_pressed("ui_up")
):
    animation.play("Walk Right")
```

Meaning that `"ui_up"` is pressed, and `"ui_right"` has been pressed for longer than `"ui_up"`.

Thus:
- This results in shorter code than my prior suggestions.
- Would only be one new method (like `is_action_held` and unlike `get_action_press_frame` and `get_action_release_frame`)
- Would be easier to reason about than my suggestion of `get_action_press_frame` and `get_action_release_frame` (with these you need to compare the release frame and press frame to know if it is pressed or released).
- You can use `Input.get_pressed_frames("action") > 1` if you wanted the `Input.is_action_held("action")` I suggested first.

Notes:
- Arguably it could be called `get_action_pressed_frames_count` but that seems too long to me.
- It could be negative for not pressed, but `0` is falsy, and if anybody does arithmetic with these without looking that it can be negative might have a surprise. So I believe `0` as not pressed is better.

**Comment by Calinou** [comment]
> Thinking about this, here is another idea: get_pressed_frames, returning an int like this:

There was a PR adding a method to return for how long an action was pressed, but it was rejected. Number of frames feels worse to me, as it's dependent on framerate or physics tick rate.

**Comment by theraot** [comment]
I see. I just gave a read it, and it seems that all the counter arguments are there.
